### ISSUE TITLE
Network.setup/2 should not return the delta 

### ISSUE THREAD

**Issue by fhunleth** [nerves-project/nerves_network#23]
Here's what I did:

```
iex> Nerves.Network.setup "wlan0", ssid: "nerves1", key_mgmt: :"WPA-PSK", psk: "..."
{:ok,
 {%{config: %{network_interface: %{"wlan0" => [ssid: "nerves1",
         key_mgmt: :"WPA-PSK", psk: "..."]}}}, %{}}}

iex> Nerves.Network.setup "wlan0", ssid: "nerves1", key_mgmt: :"WPA-PSK", psk: "..."
{:ok,
 {%{config: %{network_interface: %{"wlan0" => [ssid: "nerves1",
         key_mgmt: :"WPA-PSK", psk: "..."]}}},
  %{config: %{network_interface: %{"wlan0" => [ssid: "nerves1",
         key_mgmt: :"WPA-PSK", psk: "..."]}}}}}
```

I think that it's a feature to allow for multiple SSIDs with the hope that any one of them might associate. However, I think that calling setup a second time on the same SSID is more likely done to change the password (or something else).

**Comment by mobileoverlord** [comment]
What you are seeing is the return from SystemRegistry.update. You are receiving the pattern `{:ok, {new_reg, old_reg}}`. We could change the returned result from the call to `Nerves.Network.setup/2` to something more normal.

**Comment by fhunleth** [comment]
Ahhh. That makes so much more sense and explains a lot! 

Yes - it would be nice to switch to something more normal. I'll leave this issue open so that this can happen.

**Comment by ConnorRigby** [comment]
would it be reasonable to simply return `:ok` | `{:error reason}` if there is an error handling mech  for this?

**Comment by ConnorRigby** [comment]
also fixed with  #34
